sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -522,146 +882,339 @@ protected override async ValueTask InitializeProcessingForPartitionAsync(Partiti
+            // If the user issues a checkpoint update, the associated ownership will have its eTag updated as well, so we
+            // will fail in claiming it here, but this instance still owns it.
+
+            return Manager.ClaimOwnershipAsync(ownershipToRenew);
+        }
 
         /// <summary>
-        ///   Starts the event processor.  In case it's already running, nothing happens.
+        ///   Starts running a task responsible for receiving and processing events in the context of a specified partition.
         /// </summary>
         ///
-        /// <returns>A task to be resolved on when the operation has completed.</returns>
-        ///
-        /// <remarks>
-        ///   If overridden, the base class implementation must be explicitly called in order to make the event processor start
-        ///   running.
-        /// </remarks>
+        /// <param name="partitionId">The identifier of the Event Hub partition the task is associated with.  Events will be read only from this partition.</param>
+        /// <param name="startingPosition">The position within the partition where the task should begin reading events.</param>
+        /// <param name="maximumWaitTime">The maximum amount of time to wait to for an event to be available before emitting an empty item; if <c>null</c>, empty items will not be published.</param>
+        /// <param name="retryOptions">The set of options to use for determining whether a failed operation should be retried and, if so, the amount of time to wait between retry attempts.</param>
+        /// <param name="trackLastEnqueuedEventProperties">Indicates whether or not the task should request information on the last enqueued event on the partition associated with a given event, and track that information as events are received.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
-        /// <exception cref="EventHubsClientClosedException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
-        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsyncHandler" /> or <see cref="ProcessErrorAsyncHandler" /> set.</exception>
+        /// <returns>The running task that is currently receiving and processing events in the context of the specified partition.</returns>
         ///
-        public override Task StartAsync()
-        {
-            Argument.AssertNotClosed(IsClosed, nameof(EventProcessorClient));
-
-            lock (StartProcessorGuard)
+        private Task RunPartitionProcessingAsync(string partitionId,
+                                                 EventPosition startingPosition,
+                                                 TimeSpan? maximumWaitTime,
+                                                 EventHubsRetryOptions retryOptions,
```

Comment:
you could just pass in your retry policy and set it as a custom policy for the consumer.   That would save you a slight bit of overhead, but cost you nothing.